src/inverse/SIA2D/gradient.jl

```diff
@@ -61,34 +68,63 @@ function SIA2D_grad_batch!(θ, simulation::Inversion)
 
         result = simulation.results.simulation[i]
 
-        # Results from forward simulation
+        ## 1- Results from forward simulation
         t = result.t
         Δt = diff(t)
         H = result.H
         glacier = simulation.glaciers[i]
 
-        # Reference data
-        t_ref = glacier.thicknessData.t
-        H_ref = glacier.thicknessData.H
+        ## 2- Reference data
+
+        # Discretization for the ice thickness loss term
+        tH_ref = tdata(glacier.thicknessData)
```

Comment:
I will add a comment that adding this here does not harm even when there is no thicknessData, since this will be nothing in that case, right? Just for clarity in the code